### PR TITLE
Add Dm control

### PR DESCRIPTION
**Pull request by cjcchen**

### CODE REVIEW

File: rllab/envs/dm_control_env.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            self._viewer = None
+
+    def step(self, action):
+        time_step = self._env.step(action)
+        if time_step.reward:
+            self._total_reward += time_step.reward
+
+        return Step(flatten_observation(time_step.observation), \
+                time_step.reward, \
+                True if (time_step.step_type == StepType.LAST) else False, \
```
Comment (ryanjulian) [inline rllab/envs/dm_control_env.py]: this is the same as time_step.step_type == StepType.LAST

File: rllab/envs/dm_control_view.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pygame
```
Comment (ryanjulian) [inline rllab/envs/dm_control_view.py]: dm_control_viewer.py

File name must match the class.

File: tests/test_dmcontrol.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        dmcontrol_env.render()
+        action = action_spec.sample()
+        next_obs, reward, done, info = dmcontrol_env.step(action)
+        if done == True:
+            break
+
+    dmcontrol_env.terminate()
+
+
+for domain, task in model_list:
```
Comment (ryanjulian) [inline tests/test_dmcontrol.py]:
```python
for domain, task in dm_control.suite.ALL_TASKS:
```

File: rllab/envs/dm_control_env.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+from dm_control import suite
+from dm_control.rl.environment import StepType
+from dm_control.rl.control import flatten_observation
+
+from rllab.envs.base import Env, Step
+from rllab.envs.dm_control_view import DmControlViewer
+from rllab.core.serializable import Serializable
+from rllab.spaces.box import Box
+'''
```
Comment (ryanjulian) [inline rllab/envs/dm_control_env.py]: docstrings go inside the class

```python
class Foo:
    """
    My wonderful docstring
    """
    pass
```

File: tests/test_dmcontrol.py (hunk 3 above)
Comment (ryanjulian) [reply tests/test_dmcontrol.py]: (also means you can delete `model_list`)

**Review by ryanjulian — CHANGES_REQUESTED** [review]

File: rllab/misc/logger.py [hunk 5]
```diff
@@ -198,8 +201,10 @@ def dump_tabular(*args, **kwargs):
         # Also write to the csv files
         # This assumes that the keys in each iteration won't change!
         for tabular_fd in list(_tabular_fds.values()):
-            writer = csv.DictWriter(tabular_fd, fieldnames=list(tabular_dict.keys()))
-            if wh or (wh is None and tabular_fd not in _tabular_header_written):
```
Comment (ryanjulian) [inline rllab/misc/logger.py]: Is there a reason this file needed to be modified to get dm_control to work? Otherwise, these changes should be moved into a different PR.

Comment (cjcchen) [reply rllab/misc/logger.py]: it is just re-formated.

File: rllab/misc/logger.py [hunk 6]
```diff
@@ -198,8 +201,10 @@ def dump_tabular(*args, **kwargs):
...
+            writer = csv.DictWriter(
+                tabular_fd, fieldnames=list(tabular_dict.keys()))
+            if wh or (wh is None
+                      and tabular_fd not in _tabular_header_written):
```
Comment (cjcchen) [inline rllab/misc/logger.py]: re-formated to these lines.

File: rllab/misc/logger.py (hunk 5 above)
Comment (ryanjulian) [reply rllab/misc/logger.py]: If a change is not related to the content of a commit/pull request, it should not appear in that commit/pull request. Let me know if you want help revising the commits to remove it. I know rewriting git history can get tricky.

**Review by cjcchen — COMMENTED** [review]
I just use yapf tools to format the whole file.

I will remove this modify.

File: rllab/misc/logger.py (hunk 6 above)
Comment (ryanjulian) [reply rllab/misc/logger.py]: I'm confused. Did you need to modify the logger.py to add dm_control support? If so, you can reformat the whole file. Otherwise, just leave logger.py out.

File: tests/test_dmcontrol.py (hunk 3 above)
Comment (ryanjulian) [reply tests/test_dmcontrol.py]: why not just use `dm_control.suite.ALL_TASKS` instead of `model_list` (which is only ever used once)

**Review by ryanjulian — APPROVED** [review]
